### PR TITLE
Add API endpoints for user tokens

### PR DESCRIPTION
**Pull request by iamigo**
GET /v1/users/{key}/tokens
DELETE /v1/users/{key}/tokens/{tokenName}
GET /v1/users/{key}/tokens/{tokenName}
POST /v1/users/{key}/tokens/{tokenName}/revoke
POST /v1/users/{key}/tokens/{tokenName}/restore

### CODE REVIEW

**Comment by shriramshankar** [comment]
@iamigo For some reason,  the new line at the end of all the files are removed.  Wondering if we do not want them anymore. I need to make these changes in my PR too if this is the case.

**Comment by iamigo** [comment]
@annyhe waiting for review

File: tests/api/v1/userTokens/revokeRestore.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    .then(() => done())
+    .catch(done);
+  });
+
+  after(u.forceDelete);
+
+  it('ok', (done) => {
+    api.post(`${path}/${uname}/tokens/${tname}/revoke`)
+    .set('Authorization', '???')
+    .send({})
```
Comment (annyhe) [inline tests/api/v1/userTokens/revokeRestore.js]: why does the test accept authorization: ????

File: tests/api/v1/userTokens/utils.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const testStartTime = new Date();
+
+module.exports = {
+  fakeUserCredentials: {
+    email: 'user1@abc.com',
+    password: 'fakePasswd',
+    username: 'user1@abc.com'
+  },
+  forceDelete(done) {
+    tu.forceDelete(tu.db.Token, testStartTime)
```
Comment (annyhe) [inline tests/api/v1/userTokens/utils.js]: this and the following files look familiar: are they from another pull request?

File: tests/api/v1/userTokens/delete.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    api.delete(`${path}/${uname}/tokens/${tname}`)
+    .set('Authorization', '???')
+    .expect(constants.httpStatus.OK)
+    .end((err, res) => {
+      if (err) {
+        done(err);
+      } else {
+        expect(res.body).to.have.property('name', tname);
+        expect(res.body.isDeleted).to.not.equal(0);
+        done();
```
Comment (annyhe) [inline tests/api/v1/userTokens/delete.js]: do we expect isDeleted to === 1?

File: tests/api/v1/userTokens/revokeRestore.js (hunk 1 above)
Comment (iamigo) [inline tests/api/v1/userTokens/revokeRestore.js]: 'cuz we're not actually enforcing that a token is valid yet

File: tests/api/v1/userTokens/delete.js (hunk 3 above)
Comment (iamigo) [inline tests/api/v1/userTokens/delete.js]: when a record is soft-deleted, the "isDeleted" field will have a non-zero long int representing the timestamp of when it was soft-deleted

**Pull request merged**